src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementPageEntityUpdater.java

```diff
@@ -126,68 +117,40 @@ public void startTimer() {
   }
 
   private void startTimer(int duration) {
-    stopped = false;
     TimerConfig timerConfig = new TimerConfig();
     timerConfig.setPersistent(false);
     timerService.createSingleActionTimer(duration, timerConfig);
   }
 
-  @Override
-  public void stopTimer() {
-    stopped = true;
-  }
-  
-  @Asynchronous
-  public void onPageIdUpdateRequest(@Observes PageIdUpdateRequest event) {
-    if (!stopped) {
-      PageId pageId = event.getId();
-      
-      if (!StringUtils.equals(pageId.getSource(), ManagementConsts.IDENTIFIER_NAME)) {
-        return;
-      }
-      
-      queue.add(event);
-    }
-  }
-  
-  @Asynchronous
-  public void onPageIdRemoveRequest(@Observes PageIdRemoveRequest event) {
-    if (!stopped) {
-      PageId pageId = event.getId();
-      
-      if (!StringUtils.equals(pageId.getSource(), ManagementConsts.IDENTIFIER_NAME)) {
-        return;
-      }
-      
-      deletePage(event, pageId);
+  @Timeout
+  public void timeout(Timer timer) {
+    if (systemSettingController.isNotTestingOrTestRunning()) {
+      executeNextTask();
     }
+
+    startTimer(systemSettingController.inTestMode() ? 1000 : TIMER_INTERVAL);
   }
   
-  @Timeout
-  public void timeout(Timer timer) {
-    if (!stopped) {
-      if (systemSettingController.isNotTestingOrTestRunning()) {
-        PageIdUpdateRequest updateRequest = queue.next();
-        if (updateRequest != null) {
-          updateManagementPage(updateRequest);
-        }
+  private void executeNextTask() {
+    IdTask<PageId> task = pageIdTaskQueue.next();
+    if (task != null) {
+      if (task.getOperation() == Operation.UPDATE) {
+        updateManagementPage(task.getId(), task.getOrderIndex()); 
+      } else if (task.getOperation() == Operation.REMOVE) {
+        deleteManagementPage(task.getId());
       }
-
-      startTimer(systemSettingController.inTestMode() ? 1000 : TIMER_INTERVAL);
     }
   }
   
-  private void updateManagementPage(PageIdUpdateRequest updateRequest) {
-    OrganizationId organizationId = updateRequest.getOrganizationId();
-    PageId pageId = updateRequest.getId();
+  private void updateManagementPage(PageId managementPageId, Long orderIndex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "updateManagementPage" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)